File: frontend/src/types/scan.ts
export type ScanType = "repository" | "docker_image";
export type ScanStatus = "pending" | "running" | "completed" | "failed";
export type Severity = "critical" | "high" | "medium" | "low" | "info";

export interface Scan {
  id: string;
  project_id: string;
  scan_type: ScanType;
  status: ScanStatus;
  target: string;
  security_score: number | null;
  created_at: string;
  completed_at: string | null;
}

export interface Vulnerability {
  id: string;
  scan_id: string;
  source: string;
  severity: Severity;
  title: string;
  description: string | null;
  cve_id: string | null;
  package_name: string | null;
  file_path: string | null;
  created_at: string;
}

File: backend/app/models/scan.py
import enum
import uuid
from datetime import datetime

from sqlalchemy import Enum, ForeignKey, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class ScanType(str, enum.Enum):
    REPOSITORY = "repository"
    DOCKER_IMAGE = "docker_image"


class ScanStatus(str, enum.Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class Scan(Base):
    __tablename__ = "scans"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    project_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("projects.id"), nullable=False)
    scan_type: Mapped[ScanType] = mapped_column(Enum(ScanType), nullable=False)
    status: Mapped[ScanStatus] = mapped_column(Enum(ScanStatus), default=ScanStatus.PENDING, nullable=False)
    target: Mapped[str] = mapped_column(String(512), nullable=False)
    security_score: Mapped[int | None] = mapped_column(nullable=True)
    created_at: Mapped[datetime] = mapped_column(default=datetime.utcnow, nullable=False)
    completed_at: Mapped[datetime | None] = mapped_column(nullable=True)

    project: Mapped["Project"] = relationship(back_populates="scans")
    vulnerabilities: Mapped[list["Vulnerability"]] = relationship(
        back_populates="scan", cascade="all, delete-orphan"
    )
    reports: Mapped[list["Report"]] = relationship(back_populates="scan", cascade="all, delete-orphan")
